fabric-keybindings-v0/src/main/java/net/fabricmc/fabric/api/client/keybinding/FabricKeyBinding.java

```diff
@@ -20,27 +20,31 @@
 import net.minecraft.client.util.InputUtil;
 import net.minecraft.util.Identifier;
 
-import net.fabricmc.fabric.mixin.client.keybinding.KeyCodeAccessor;
-
 /**
  * Expanded version of {@link KeyBinding} for use by Fabric mods.
  *
  * <p>*ALL* instantiated FabricKeyBindings should be registered in
- * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
+ * {@link KeyBindingRegistry#register(FabricKeyBinding)}!</p>
+ *
+ * @deprecated Please migrate to v1. Please use {@link net.fabricmc.fabric.api.client.keybinding.v1.FabricKeyBinding} instead.
  */
-public class FabricKeyBinding extends KeyBinding {
+@Deprecated
+public class FabricKeyBinding extends net.fabricmc.fabric.api.client.keybinding.v1.FabricKeyBinding {
 	protected FabricKeyBinding(Identifier id, InputUtil.Type type, int code, String category) {
-		super("key." + id.toString().replace(':', '.'), type, code, category);
+		super(id, String.format("key.%s.%s", id.getNamespace(), id.getPath()), type, code, category);
 	}
 
 	/**
 	 * Returns the configured KeyCode assigned to the KeyBinding from the player's settings.
+	 *
 	 * @return configured KeyCode
 	 */
+	@Override
 	public InputUtil.KeyCode getBoundKey() {
-		return ((KeyCodeAccessor) this).getKeyCode();
+		return super.getBoundKey();
```

Comment:
Can probably remove this override?

Reply:
will it preserve backwards compatibility if I do this? (I am not sure so I did this)

Reply:
Make the v0 use a delegate class internally like how I did FabricBlockSettings v1

Reply:
but the v0 one extends Keybinding anyways

Reply:
It should preserve backward compatibility... the call to old keybinding's `getBoundKey` should be invokevirtual in existing bytecode

Hmm, I guess we can give the new method a new name like `getDefaultCode` instead

Reply:
is not the default code, it returns the current code, bound key is the best name for this imo